test/util/test_runners/integration_test_case.py

```diff
@@ -0,0 +1,89 @@
+"""
+extend the unitest TestCase class
+"""
+
+# Builtins
+import io
+import sys
+
+# Third party
+import itertools
+from unittest import TestCase
+from contextlib import redirect_stdout
+import subprocess as sp
+from unittest.mock import patch, MagicMock
+
+# Internals
+from conjur.cli import Cli
+from test.util.test_runners.params import ClientParams, TestEnvironmentParams
+
+
+class IntegrationTestCaseBase(TestCase):
+
+    def __init__(self, testname, client_params: ClientParams = None,
+                 environment_params: TestEnvironmentParams = None):
+        """
+        Base class that extends unittest TestCase
+        used to setup tests environment
+        """
+        self.environment = TestEnvironmentParams() if environment_params is None else environment_params
+        self.client_params = ClientParams() if client_params is None else client_params
+
+        super(IntegrationTestCaseBase, self).__init__(testname)
+
+
+    def invoke_cli(self, *args, exit_code=0) -> str:
+        """
+        Invoker for the integration tests.
+        if self.environment.invoke_process == True
+        than it will call the cli binaries to run the integration tests
+        else it will use the Cli code to invoke
+        """
+        if self.environment.invoke_process:
+            return invoke_cli_as_process(self, *args, exit_code=exit_code)
+        return invoke_cli_as_code(self, *args, exit_code=exit_code)
+
+
+def invoke_cli_as_code(test_runner, *args, exit_code=0):
+    """
+        Invoke cli command using python code as seen below.
+        The usecase of this is when we run the tests
+        on development environment
```

Comment:
not sure what you mean by development environment here.
do you mean containerized test environment?

Reply:
when you run from python ide or run it as python scrupt. adding